client/router_create.go

```diff
@@ -624,22 +626,60 @@ func (cli *VanClient) GetRouterSpecFromOpts(options types.SiteConfigSpec, siteId
 	})
 
 	if !isEdge {
-		credentials = append(credentials, types.Credential{
-			CA:          types.SiteCaSecret,
-			Name:        types.SiteServerSecret,
-			Subject:     types.TransportServiceName,
-			Hosts:       []string{types.TransportServiceName + "." + van.Namespace},
-			ConnectJson: false,
-			Post:        !options.IsIngressNone(),
-		})
-		van.ControllerCredentials = append(van.ControllerCredentials, types.Credential{
-			CA:          types.SiteCaSecret,
-			Name:        types.ClaimsServerSecret,
-			Subject:     types.ControllerServiceName,
-			Hosts:       []string{types.ControllerServiceName + "." + van.Namespace},
-			ConnectJson: false,
-			Post:        !options.IsIngressNone(),
-		})
+		if options.IsIngressNone() {
+			credentials = append(credentials, types.Credential{
+				CA:          types.SiteCaSecret,
+				Name:        types.SiteServerSecret,
+				Subject:     types.TransportServiceName,
+				Hosts:       []string{types.TransportServiceName + "." + van.Namespace},
+				ConnectJson: false,
+				Post:        false,
+			})
+			van.ControllerCredentials = append(van.ControllerCredentials, types.Credential{
+				CA:          types.SiteCaSecret,
+				Name:        types.ClaimsServerSecret,
+				Subject:     types.ControllerServiceName,
+				Hosts:       []string{types.ControllerServiceName + "." + van.Namespace},
+				ConnectJson: false,
+				Post:        false,
+			})
+		} else if options.IsIngressNodePort() {
+			credentials = append(credentials, types.Credential{
+				CA:          types.SiteCaSecret,
+				Name:        types.SiteServerSecret,
+				Subject:     options.Router.IngressHost,
+				Hosts:       []string{types.TransportServiceName + "." + van.Namespace},
+				ConnectJson: false,
+				Post:        false,
+			})
+			if options.Controller.IngressHost != "" {
```

Comment:
Would this work if rather than options.Controller.IngressHost,  options.Router.IngressHost is used for the claims secret? 

I tried `skupper init --ingress nodeport --router-ingress-host 192.168.39.231` and it failed as claims secret did not get created. When i tried  `skupper init --ingress nodeport --router-ingress-host 192.168.39.231 --controller-ingress-host 192.168.39.231` it worked fine.

It seems reasonable that claims service would  correlate to router-ingress rather than console-ingress if that makes sense.

Reply:
The claims are processed by the controller at present, therefore the host used has to be of the node on which the controller is running. The later PR that includes nginx ingress adds a --ingress-host option which can also be used for nodeport in the event that both controller and router are running on the same node.